clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -1721,6 +1721,40 @@ static void CollectARMPACBTIOptions(const ToolChain &TC, const ArgList &Args,
   }
 }
 
+static void EmitAccuracyDiag(const Driver &D, const JobAction &JA,
+                             StringRef AccuracValStr, StringRef TargetPrecStr) {
+  if (JA.isDeviceOffloading(Action::OFK_SYCL)) {
+    D.Diag(clang::diag::warn_acuracy_conflicts_with_explicit_target_prec_option)
+        << AccuracValStr << TargetPrecStr;
+  }
+}
+
+void Clang::AddSPIRTargetArgs(const ArgList &Args, ArgStringList &CmdArgs,
+                              const JobAction &JA, const Driver &D) const {
+  if (JA.isDeviceOffloading(Action::OFK_SYCL)) {
+    if (Arg *A = Args.getLastArg(options::OPT_ffp_model_EQ)) {
+      if (!strcmp(A->getValue(), "fast")) {
+        CmdArgs.push_back("-fno-offload-fp32-prec-div");
+        CmdArgs.push_back("-fno-offload-fp32-prec-sqrt");
+      }
+    }
+    if (Arg *A = Args.getLastArg(options::OPT_ffp_accuracy_EQ)) {
+      if (Args.getLastArg(options::OPT_fno_offload_fp32_prec_div))
+        EmitAccuracyDiag(D, JA, A->getValue(), "-fno-offload-fp32-prec-div");
+      if (Args.getLastArg(options::OPT_fno_offload_fp32_prec_sqrt))
+        EmitAccuracyDiag(D, JA, A->getValue(), "-fno-offload-fp32-prec-sqrt");
+    }
+    if (Args.getLastArg(options::OPT_fno_offload_fp32_prec_div))
+      CmdArgs.push_back("-fno-offload-fp32-prec-div");
+    else
+      CmdArgs.push_back("-foffload-fp32-prec-div");
```

Comment:
```suggestion
    if (!Args.hasFlag(option::OPT_foffload_fp32_prec_div,
                      option::OPT_fno_offload_fp32_prec_div, true))
      CmdArgs.push_back("-fno-offload-fp32-prec-div");
```
Since `-foffload-fp32-prec-div` is default